src/OpenTelemetry.Shared/PersistentStorageHelper.cs

```diff
@@ -0,0 +1,223 @@
+        }
+
+        internal static void RemoveExpiredBlobs(string directoryPath, long retentionPeriodInMilliseconds, long writeTimeoutInMilliseconds)
+        {
+            var currentUtcDateTime = DateTime.Now.ToUniversalTime();
+
+            var leaseDeadline = currentUtcDateTime;
+            var retentionDeadline = currentUtcDateTime - TimeSpan.FromMilliseconds(retentionPeriodInMilliseconds);
+            var timeoutDeadline = currentUtcDateTime - TimeSpan.FromMilliseconds(writeTimeoutInMilliseconds);
+
+            foreach (var file in Directory.GetFiles(directoryPath).OrderByDescending(f => f))
+            {
+                var success = RemoveTimedOutTmpFiles(timeoutDeadline, file);
+
+                if (success)
+                {
+                    continue;
+                }
+
+                success = RemoveExpiredLease(leaseDeadline, file);
+
+                if (!success)
+                {
+                    RemoveExpiredBlob(retentionDeadline, file);
+                }
+            }
+        }
+
+        internal static string GetUniqueFileName(string extension)
+        {
+            return string.Format(CultureInfo.InvariantCulture, $"{DateTime.Now.ToUniversalTime():yyy-MM-ddTHHmmss.ffffff}-{Guid.NewGuid().ToString("N", CultureInfo.InvariantCulture)}{extension}");
+        }
+
+        internal static string CreateSubdirectory(string path)
+        {
+            string subdirectoryPath = string.Empty;
+
+            try
+            {
+                string baseDirectory = string.Empty;
+#if !NETSTANDARD
+                baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+#else
+                baseDirectory = AppContext.BaseDirectory;
+#endif
+
+                string appIdentity = Environment.UserName + "@" + Path.Combine(baseDirectory, Process.GetCurrentProcess().ProcessName);
+                string subdirectoryName = GetSHA256Hash(appIdentity);
+                subdirectoryPath = Path.Combine(path, subdirectoryName);
+                Directory.CreateDirectory(subdirectoryPath);
+            }
+            catch (Exception ex)
+            {
+                SharedEventSource.Log.Error($"Error creating sub-directory {path}.", ex);
+            }
+
+            return subdirectoryPath;
+        }
+
+        internal static void DeleteFile(string path)
+        {
+            try
+            {
+                if (File.Exists(path))
+                {
+                    File.Delete(path);
+                }
+            }
+            catch (Exception ex)
+            {
+                SharedEventSource.Log.Warning($"Deletion of file {path} has failed.", ex);
+            }
+        }
+
+        internal static float CalculateFolderSize(string path)
+        {
+            if (!Directory.Exists(path))
+            {
+                return 0;
+            }
+
+            float directorySize = 0.0f;
+            try
+            {
+                foreach (string file in Directory.GetFiles(path))
```

Comment:
not sure if it is worth, but we could change from GetFiles to EnumerateFiles (not sure how many files we will have).